shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -133,6 +133,34 @@ FLUTTER_EXPORT
  */
 - (void)destroyContext;
 
+/**
+ * Ensures that Flutter will generate a semantics tree.
+ *
+ * This is enabled by default if certain accessibility services are turned on by
+ * the user, or when using a Simulator. This method allows a user to turn
+ * semantics on when they would not ordinarily be generated and the performance
+ * overhead is not a concern, e.g. for UI testing. Note that semantics should
+ * never be programatically turned off, as it would potentially disable
+ * accessibility services an end user has requested.
+ *
+ * This method must only be called after launching the engine via
+ * `-runWithEntrypoint:` or `-runWithEntryPoint:libraryURI`.
```

Comment:
Let me know if I'm wrong: In the perspective FlutterViewController user, it's hard to find a right time to enable semantics before the first semantics tree being built.

I think it will be easier if FlutterViewController can take the preference and invoke this method when _engineNeedsLaunch is used :).

Reply:
Semantics might already be enabled based on device settings or if it's a simulator.  It should be safe to call this method even if semantics are enabled - if they're not, no semantics tree will be built.

We could make it an option in the initializer, but I'm not sure how that would help for testing.

Reply:
Or maybe a property/one-way-method of FlutterViewController? User can check if they need to force enabling semantics through several ways, such as plist or envs. If it's the case, they can flip the bit after calling initilizer, but before load the VC.

Reply:
The `ensureSemanticsEnabled` is safe to call at any time.  Basically, the idea is to just call it right after you register for the callback.  You could in theory call it in a loop after that as long as you let the main runloop make progress.